File: py/ege9.144v2.py
f = open("ege9.144.txt")

n = 0

for l in f:
    if l == '\n': continue

    ax = list(map(int, l.split()))

    if all(1 <= x <= 8 for x in ax):
        x1, y1, x2, y2 = ax
        ad = [x1 - x2, y1 - y2]
        if sum(x ** 2 for x in ad) == 5:
            n += 1

print(n)

File: c/ege9.144v2.c
#include <stdio.h>
#include "array.lib.c"

int main() {
    FILE *f = fopen("ege9.144.txt", "r");
    int ax[4];
    int n = 0;

    while(freadn(f, ax, 4) == 4) {
        if(min(ax, 4) < 1 || max(ax, 4) > 8) continue;

        int ad[] = {ax[0] - ax[2], ax[1] - ax[3]};
        if(sumsqr(ad, 2) == 5) { 
            n++;
        }
    }
    printf("%d\n", n);
}
